clients/src/main/java/org/apache/kafka/common/network/ServerConnectionId.java

```diff
@@ -0,0 +1,143 @@
+        int processorId,
+        int index
+    ) {
+        this(localEndpoint.host(), localEndpoint.port(), remoteEndpoint.host(), remoteEndpoint.port(), processorId, index);
+    }
+
+    public ServerConnectionId(
+        String localHost,
+        int localPort,
+        String remoteHost,
+        int remotePort,
+        int processorId,
+        int index
+    ) {
+        this.localHost = localHost;
+        this.localPort = localPort;
+        this.remoteHost = remoteHost;
+        this.remotePort = remotePort;
+        this.processorId = processorId;
+        this.index = index;
+    }
+
+    public String localHost() {
+        return localHost;
+    }
+
+    public int localPort() {
+        return localPort;
+    }
+
+    public String remoteHost() {
+        return remoteHost;
+    }
+
+    public int remotePort() {
+        return remotePort;
+    }
+
+    public int processorId() {
+        return processorId;
+    }
+
+    public int index() {
+        return index;
+    }
+
+    /**
+     * Returns an optional ServerConnectionId object from the given connection ID string.
+     *
+     * @param connectionIdString The connection ID string to parse.
+     * @return An optional ServerConnectionId object.
+     */
+    public static Optional<ServerConnectionId> fromString(String connectionIdString) {
+        String[] split = connectionIdString.split("-");
+        if (split.length != 4) {
+            return Optional.empty();
+        }
+
+        try {
+            return parseHostPort(split[0]).flatMap(localHost -> parseHostPort(split[1]).map(
+                remoteHost -> new ServerConnectionId(localHost, remoteHost, Integer.parseInt(split[2]), Integer.parseInt(split[3]))));
+        } catch (NumberFormatException e) {
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * Generates a unique connection ID for the given socket.
+     *
+     * @param socket The socket for which the connection ID is to be generated.
+     * @param processorId The ID of the server processor that will handle this connection.
+     * @param connectionIndex The index to be used in the connection ID to ensure uniqueness.
+     * @return A string representing the unique connection ID.
+     */
+    public static String generateConnectionId(Socket socket, int processorId, int connectionIndex) {
+        String localHost = socket.getLocalAddress().getHostAddress();
+        int localPort = socket.getLocalPort();
+        String remoteHost = socket.getInetAddress().getHostAddress();
+        int remotePort = socket.getPort();
+        return localHost + ":" + localPort + "-" + remoteHost + ":" + remotePort + "-" + processorId + "-" + connectionIndex;
+    }
+
+    /**
+     * BrokerEndPoint URI is host:port or [ipv6_host]:port
+     * Note that unlike EndPoint (or listener) this URI has no security information.
+     */
+    // Visible for testing
+    static Optional<BrokerEndPoint> parseHostPort(String connectionString) {
```

Comment:
Could you please replace `BrokerEndPoint` by `Map.Entry`? we are under java 11, so it is easier to create `Entry` now.

Reply:
Done, though not a big fan of Map.Entry but seems sensible here as there was no broker id for BrokerEndpoint.